src/Compute/custom/NewAzVM_SimpleParameterSet.cs

```diff
@@ -0,0 +1,346 @@
+        public SwitchParameter SystemAssignedIdentity { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = "Use this to add the assign user specified identity (MSI) to the VM")]
+        [ValidateNotNullOrEmpty]
+        public string UserAssignedIdentity { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = "Run cmdlet in the background")]
+        public SwitchParameter AsJob { get; set; }
+
+        [Parameter(Mandatory = false)]
+        public int[] DataDiskSizeInGb { get; set; }
+
+        [Parameter(Mandatory = false)]
+        public SwitchParameter EnableUltraSSD { get; set; }
+
+        [Parameter(Mandatory = false)]
+        public string ProximityPlacementGroup { get; set; }
+
+        public CancellationToken Token => Source.Token;
+
+        public Action Cancel => Source.Cancel;
+
+        protected override void StopProcessing()
+        {
+            this.Cancel();
+            base.StopProcessing();
+        }
+
+        public override void ExecuteCmdlet()
+        {
+            base.ExecuteCmdlet();
+            this.StartAndWait(StrategyExecuteCmdletAsync);
+        }
+
+        class Parameters : IParameters<VirtualMachine>
+        {
+            readonly NewAzVM_SimpleParameterSet _cmdlet;
+
+            readonly Client _client;
+
+            readonly IResourceManagementClient _resourceClient;
+
+            public Parameters(NewAzVM_SimpleParameterSet cmdlet, Client client, IResourceManagementClient resourceClient)
+            {
+                _cmdlet = cmdlet;
+                _client = client;
+                _resourceClient = resourceClient;
+            }
+
+            public ImageAndOsType ImageAndOsType { get; set; }
+
+            public string Location
+            {
+                get { return _cmdlet.Location; }
+                set { _cmdlet.Location = value; }
+            }
+
+            string _defaultLocation = null;
+
+            public string DefaultLocation
+            {
+                get
+                {
+                    if (_defaultLocation == null)
+                    {
+                        var vmResourceType = _resourceClient.Providers.GetAsync("Microsoft.Compute").ConfigureAwait(false).GetAwaiter().GetResult()
+                        .ResourceTypes.Where(a => String.Equals(a.ResourceType, "virtualMachines", StringComparison.OrdinalIgnoreCase))
+                                      .FirstOrDefault();
+                        if (vmResourceType != null)
+                        {
+                            var availableLocations = vmResourceType.Locations.Select(a => a.ToLower().Replace(" ", ""));
+                            if (availableLocations.Any(a => a.Equals("eastus")))
+                            {
+                                _defaultLocation = "eastus";
+                            }
+                            _defaultLocation = availableLocations.FirstOrDefault() ?? "eastus";
```

Comment:
@markcowl shouldn't we put this line in the `else` block for the above if-statement? If the `availableLocations` list has an "eastus" entry, but it's not first, then it's going to be overwritten on this line. I think we should be doing the following:

```cs
if (availableLocations.Any(a => a.Equals("eastus")))
{
    _defaultLocation = "eastus";
}
else
{
    _defaultLocation = availableLocations.FirstOrDefault() ?? "eastus";
}
```